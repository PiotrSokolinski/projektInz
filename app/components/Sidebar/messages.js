/*
 * Navbar Messages
 *
 * This contains all the text for the Navbar component.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.components.Navbar'

export default defineMessages({
  home: {
    id: `${scope}.home`,
    defaultMessage: 'Home',
  },
  tasks: {
    id: `${scope}.tasks`,
    defaultMessage: 'Tasks',
  },
  calendar: {
    id: `${scope}.calendar`,
    defaultMessage: 'Calendar',
  },
  settings: {
    id: `${scope}.settings`,
    defaultMessage: 'Settings',
  },
})
