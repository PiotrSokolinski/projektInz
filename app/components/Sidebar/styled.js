import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { Home } from 'styled-icons/fa-solid/Home'
import { Task } from 'styled-icons/boxicons-regular/Task'
import { Calendar } from 'styled-icons/boxicons-regular/Calendar'
import { Settings } from 'styled-icons/material/Settings'
import { Colors } from 'themes'

export const Container = styled.div`
  height: 100vh;
  transition: width 0.5s;
  width: ${props => (props.isSidebarOpen ? '200px' : '60px')};
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  overflow: hidden;
  border-right: 1px solid ${Colors.silverChalice};
  background: ${Colors.aquaHaze};
  a:first-of-type {
    margin-top: 25px;
  }
`

export const HamburgerMenu = styled.div`
  cursor: pointer;
  width: 30px;
  margin-left: 15px;
  margin-top: 8px;
  z-index: 1001;
  &::after,
  &::before,
  div {
    background-color: ${Colors.white};
    border-radius: 2px;
    content: '';
    display: block;
    height: 4px;
    margin: 5px 0;
    transition: all 0.2s ease-in-out;
  }
  &::after {
    ${props => (props.isSidebarOpen ? 'transform: translateY(-6px) rotate(-135deg);' : '')};
  }
  &::before {
    ${props => (props.isSidebarOpen ? 'transform: translateY(12px) rotate(135deg);' : '')};
  }
  div {
    ${props => (props.isSidebarOpen ? 'transform: scale(0);' : '')};
  }
`

export const LinkElement = styled(Link)`
  display: flex;
  align-items: center;
  cursor: pointer;
  margin-top: 25px;
  margin-left: 8px;
`
export const Text = styled.div`
  color: ${Colors.black};
  margin-left: 15px;
  ${props => (props.isActive ? `color: ${Colors.cerulean};` : '')};
`

export const HomeIcon = styled(Home)`
  color: ${Colors.black};
  ${props => (props.isActive ? `color: ${Colors.cerulean};` : '')};
`
export const TaskIcon = styled(Task)`
  color: ${Colors.black};
  ${props => (props.isActive ? `color: ${Colors.cerulean};` : '')};
`

export const CalendarIcon = styled(Calendar)`
  color: ${Colors.black};
  ${props => (props.isActive ? `color: ${Colors.cerulean};` : '')};
`
export const SettingsIcon = styled(Settings)`
  color: ${Colors.black};
  ${props => (props.isActive ? `color: ${Colors.cerulean};` : '')};
`
