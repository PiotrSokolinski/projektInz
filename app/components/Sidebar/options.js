import React from 'react'
import { FormattedMessage } from 'react-intl'
import messages from './messages'
import { HomeIcon, TaskIcon, CalendarIcon, SettingsIcon } from './styled'

export const options = [
  {
    to: '/',
    label: <FormattedMessage {...messages.home} />,
    Icon: HomeIcon,
  },
  {
    to: '/tasks',
    label: <FormattedMessage {...messages.tasks} />,
    Icon: TaskIcon,
  },
  {
    to: '/calendar',
    label: <FormattedMessage {...messages.calendar} />,
    Icon: CalendarIcon,
  },
  {
    to: '/settings',
    label: <FormattedMessage {...messages.settings} />,
    Icon: SettingsIcon,
  },
]
