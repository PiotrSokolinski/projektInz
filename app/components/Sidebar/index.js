/**
 *
 * Sidebar
 *
 */

import React, { useState } from 'react'
import map from 'lodash/map'
// import PropTypes from 'prop-types';
import * as Styled from './styled'
import { options } from './options'

const Sidebar = () => {
  const [isSidebarOpen, setIsSidebarOpen] = useState(false)
  return (
    <Styled.Container isSidebarOpen={isSidebarOpen}>
      <Styled.HamburgerMenu isSidebarOpen={isSidebarOpen} onClick={() => setIsSidebarOpen(!isSidebarOpen)}>
        <div></div>
      </Styled.HamburgerMenu>
      {map(options, option => {
        const isActive = option.to === window.location.pathname
        return (
          <Styled.LinkElement to={option.to} key={`el-option-${option.to}`}>
            <option.Icon size="40" isActive={isActive} />
            <Styled.Text isActive={isActive}>{option.label}</Styled.Text>
          </Styled.LinkElement>
        )
      })}
    </Styled.Container>
  )
}

Sidebar.propTypes = {}

export default Sidebar
