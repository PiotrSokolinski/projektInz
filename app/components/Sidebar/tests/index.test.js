/**
 *
 * Tests for Sidebar
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { IntlProvider } from 'react-intl'

import Sidebar from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<Sidebar />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MemoryRouter>
          <Sidebar />
        </MemoryRouter>
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MemoryRouter>
          <Sidebar />
        </MemoryRouter>
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
