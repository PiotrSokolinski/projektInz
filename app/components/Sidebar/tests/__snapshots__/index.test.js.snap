// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> Should render and match the snapshot 1`] = `
<div
  class="styled__Container-sc-1etlmco-0 dMzpuT"
>
  <div
    class="styled__HamburgerMenu-sc-1etlmco-1 kFoaQf"
  >
    <div />
  </div>
  <a
    class="styled__LinkElement-sc-1etlmco-2 dEIZoa"
    href="/"
  >
    <svg
      aria-hidden="true"
      class="styled__HomeIcon-sc-1etlmco-4 lVLeK StyledIconBase-sc-bdy9j4 cCvWXa"
      fill="currentColor"
      focusable="false"
      height="40"
      viewBox="0 0 576 512"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M280.37 148.26L96 300.11V464a16 16 0 0016 16l112.06-.29a16 16 0 0015.92-16V368a16 16 0 0116-16h64a16 16 0 0116 16v95.64a16 16 0 0016 16.05L464 480a16 16 0 0016-16V300L295.67 148.26a12.19 12.19 0 00-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 00-12-12h-56a12 12 0 00-12 12v72.61L318.47 43a48 48 0 00-61 0L4.34 251.47a12 12 0 00-1.6 16.9l25.5 31A12 12 0 0045.15 301l235.22-193.74a12.19 12.19 0 0115.3 0L530.9 301a12 12 0 0016.9-1.6l25.5-31a12 12 0 00-1.7-16.93z"
        fill="currentColor"
      />
    </svg>
    <div
      class="styled__Text-sc-1etlmco-3 jqoBkU"
    >
      <span>
        Home
      </span>
    </div>
  </a>
  <a
    class="styled__LinkElement-sc-1etlmco-2 dEIZoa"
    href="/tasks"
  >
    <svg
      aria-hidden="true"
      class="styled__TaskIcon-sc-1etlmco-5 gkVBSN StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      height="40"
      viewBox="0 0 24 24"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 22h14c1.103 0 2-.897 2-2V5c0-1.103-.897-2-2-2h-3a1 1 0 00-1-1H9a1 1 0 00-1 1H5c-1.103 0-2 .897-2 2v15c0 1.103.897 2 2 2zM5 5h3v2h8V5h3v15H5V5z"
      />
      <path
        d="M11 13.586l-1.793-1.793-1.414 1.414L11 16.414l5.207-5.207-1.414-1.414z"
      />
    </svg>
    <div
      class="styled__Text-sc-1etlmco-3 cOwkHa"
    >
      <span>
        Tasks
      </span>
    </div>
  </a>
  <a
    class="styled__LinkElement-sc-1etlmco-2 dEIZoa"
    href="/calendar"
  >
    <svg
      aria-hidden="true"
      class="styled__CalendarIcon-sc-1etlmco-6 cZVpdc StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      height="40"
      viewBox="0 0 24 24"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 11h2v2H7zm0 4h2v2H7zm4-4h2v2h-2zm0 4h2v2h-2zm4-4h2v2h-2zm0 4h2v2h-2z"
      />
      <path
        d="M5 22h14c1.103 0 2-.897 2-2V6c0-1.103-.897-2-2-2h-2V2h-2v2H9V2H7v2H5c-1.103 0-2 .897-2 2v14c0 1.103.897 2 2 2zM19 8l.001 12H5V8h14z"
      />
    </svg>
    <div
      class="styled__Text-sc-1etlmco-3 cOwkHa"
    >
      <span>
        Calendar
      </span>
    </div>
  </a>
  <a
    class="styled__LinkElement-sc-1etlmco-2 dEIZoa"
    href="/settings"
  >
    <svg
      aria-hidden="true"
      class="styled__SettingsIcon-sc-1etlmco-7 eiqXfJ StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      height="40"
      viewBox="0 0 24 24"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.43 12.98c.04-.32.07-.64.07-.98s-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.12-.22-.39-.3-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65A.488.488 0 0014 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.23-.09-.49 0-.61.22l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98s.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.12.22.39.3.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.23.09.49 0 .61-.22l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
      />
    </svg>
    <div
      class="styled__Text-sc-1etlmco-3 cOwkHa"
    >
      <span>
        Settings
      </span>
    </div>
  </a>
</div>
`;
